presto-main/src/main/java/com/facebook/presto/server/SerializedPageWriteListener.java

```diff
@@ -52,7 +51,7 @@ public SerializedPageWriteListener(
         this.serializedPages = new ArrayDeque<>(requireNonNull(serializedPages, "serializedPages is null"));
         this.asyncContext = requireNonNull(asyncContext, "asyncContext is null");
         this.output = requireNonNull(output, "output is null");
-        this.slice = allocate(PAGE_METADATA_SIZE);
+        this.pageMetadataSliceOutput = allocate(PAGE_METADATA_SIZE).getOutput();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *pageMetadataSliceOutput*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).